### DISCUSSION TITLE
More appropriate error descriptions

### DISCUSSION THREAD

**Discussion by bojasv** [codefori/vscode-ibmi#1260] [General]
Dear folks, 

I'd like to congratulate the contributors for the new release. It has many cool features incl. the helpful error assist while adding a new member. While I was working with it I found a minor correction that could be made for better usability and I'd like to discuss the same here. 

Below is an occurrence where I found that the following error could be highlighted more aptly, maybe saying 
"Error: Source Member name length can't exceed 10 characters.". 

![image](https://user-images.githubusercontent.com/59419054/235101038-48556804-26e5-40aa-b9c4-cdeb6f1d0745.png)

I checked settings.json and searched the repo for error messages but couldn't found any references/sources that could help me to suggest my changes thru a PR. I'm gradually learning Git, if this is something that looks a considerable issue and can be fixed quickly, please do the needful or guide me how can I add this change ;) 

Cheers!

**Comment by worksofliam** [comment]
Hi @bojasv 

There isn't a real bug here. It's true, when creating source members, the name can only be a maximum of ten characters due to limitations of the QSYS file system.

**Reply by bojasv** [reply]
Hello @worksofliam, Thank you for your response. 

That's correct this is something which isn't an actual performance bug. In my view it's good to provide users more appropriate help. 

![image](https://user-images.githubusercontent.com/59419054/235282031-c0d0cd79-7c6a-4db0-bf49-7eba5f1eab28.png)

As in above example, for majority of IBM i users who still use green screen, the OS actually highlight the fields and provide relevant messages.
hence it becomes an easy find that you'd need to bring name length down to 10 and also to name a SRCMBR starting with sp-char only few of them (@, # and $) can be used and '&' isn't one of them.
On the other hand, considering growing IBM i newbies who find Code for i to be their preferred IDE, this is quite vague, and if corrected it'd add good value to their colorful IBM i xp ;)

I could push a suggestive correction if I could be helped with the relevant code-pieces where such errors are being handled.
